### ISSUE TITLE
Can‘t reproduce the same test results on BSD68 using my own USRNET training model.

### ISSUE THREAD

**Issue by pigfather0315** [cszn/KAIR#50]
Hi, I trained the USRNet using the code in this repository recently. I haven't found problems in data and network during training. But I got the worse results than yours in the paper. So I'd like to ask for some hints to train the model correctly.
I was wondering whether the mannual seed affecting the results. If so, Can you show me your setting of mannual seed during your training?  Thanks for your help.

![微信图片_20201104101808](https://user-images.githubusercontent.com/60414736/98062404-ee759180-1e88-11eb-8ac3-68556b00687a.png)
My results were shown above, the blue ones are results in the paper. I can get the same results using the pretrained model download from drive. The red ones are my results which have a large gap between the ones in papers.

**Comment by cszn** [comment]
Set https://github.com/cszn/KAIR/blob/5254362bcd00f62bc1de92d9289377814e4990d4/options/train_usrnet.json#L43
with [64, 128, 256, 512].

Increase https://github.com/cszn/KAIR/blob/5254362bcd00f62bc1de92d9289377814e4990d4/options/train_usrnet.json#L22
or modify https://github.com/cszn/KAIR/blob/5254362bcd00f62bc1de92d9289377814e4990d4/options/train_usrnet.json#L67 because I use 4 GPUs for training.

**Comment by pigfather0315** [comment]
@cszn Thanks for your reply. I have set nc with [64, 128, 256, 512]. As for the batch_size, I set it with 24 due to the limited GPU memory. Lastly, should I set the “G_scheduler_milestones” with [400000, 800000, 1200000, 1600000] using one GPU for training?

**Comment by cszn** [comment]
Yes, just more training.

**Comment by pigfather0315** [comment]
@cszn Thanks a lot. I will try it again in this way.

**Comment by clttyou** [comment]
> @cszn Thanks a lot. I will try it again in this way.

Hi,I encountered the same problem (can‘t reproduce the same test results on BSD68 using my own USRNET training model.). The current learning rate has reached 3.125*10^(-6), but the test results in the paper are still not available. How do you solve this problem? Thank you very much. I use 4 GPUs for training,my configuration is as follows：
{
  "task": "usrnet" //  
  , "model": "plain4" // "plain" | "gan"
  , "gpu_ids": [0,1,2,3]
  , "scale": 4
  , "n_channels": 3 // 1 for grayscale image restoration, 3 for color image restoration
  , "merge_bn": false
  , "merge_bn_startpoint": 300000

  , "datasets": {
    "train": {
      "name": "train_dataset"
      , "dataset_type": "usrnet"
      , "dataroot_H": "trainsets/trainH"
      , "dataroot_L": null
      , "H_size": 96 // 128 | 192
      , "use_flip": true
      , "use_rot": true
      , "scales": [1, 2, 3, 4]
      , "dataloader_shuffle": true
      , "dataloader_num_workers": 8
      , "dataloader_batch_size": 96
    }
    , "test": {
      "name": "test_dataset"
      , "dataset_type": "usrnet"
      , "dataroot_H": "testsets/set5"
      , "dataroot_L": null
    }
  }

  , "path": {
    "root": "SR"
    , "pretrained_netG": null
  }

  , "netG": {
    "net_type": "usrnet" // "srresnet" | "rrdbnet" | "rcan" | "unet" | "unetplus" | "nonlocalunet"
    , "n_iter": 8 // 6
    , "h_nc": 64 // 32
    , "in_nc": 4
    , "out_nc": 3
    , "nc": [64, 128, 256, 512] // for "usrnet_tiny" : [16, 32, 64, 64]
    , "nb": 2
    , "gc": 32
    , "ng": 2
    , "reduction" : 16
    , "act_mode": "R" // "BR" for BN+ReLU | "R" for ReLU
    , "upsample_mode": "convtranspose" // "pixelshuffle" | "convtranspose" | "upconv"
    , "downsample_mode": "strideconv" // "strideconv" | "avgpool" | "maxpool"

    , "init_type": "orthogonal" // "orthogonal" | "normal" | "uniform" | "xavier_normal" | "xavier_uniform" | "kaiming_normal" | "kaiming_uniform"
    , "init_bn_type": "uniform" // "uniform" | "constant"
    , "init_gain": 0.2
  }

  , "train": {
    "G_lossfn_type": "l1" // "l1" | "l2sum" | "l2" | "ssim"
    , "G_lossfn_weight": 1.0

    , "G_optimizer_type": "adam"
    , "G_optimizer_lr": 1e-4
    , "G_optimizer_wd": 0
    , "G_optimizer_clipgrad": null

    , "G_scheduler_type": "MultiStepLR"
    , "G_scheduler_milestones": [40000, 80000, 120000, 160000,200000]
    , "G_scheduler_gamma": 0.5

    , "G_regularizer_orthstep": null
    , "G_regularizer_clipstep": null

    , "checkpoint_test": 5000
    , "checkpoint_save": 5000
    , "checkpoint_print": 200
  }
}
